ratelimits/limiter.go

```diff
@@ -118,94 +118,81 @@ type Decision struct {
 	newTAT time.Time
 }
 
-// Check returns a *Decision that indicates whether there's enough capacity to
-// allow the request, given the cost, for the specified limit Name and client
-// id. However, it DOES NOT deduct the cost of the request from the bucket's
-// capacity. Hence, the returned *Decision represents the hypothetical state of
-// the bucket if the cost WERE to be deducted. The returned *Decision will
-// always include the number of remaining requests in the bucket, the required
-// wait time before the client can make another request, and the time until the
-// bucket refills to its maximum capacity (resets). If no bucket exists for the
-// given limit Name and client id, a new one will be created WITHOUT the
-// request's cost deducted from its initial capacity. If the specified limit is
-// disabled, ErrLimitDisabled is returned.
-func (l *Limiter) Check(ctx context.Context, name Name, id string, cost int64) (*Decision, error) {
-	if cost < 0 {
+// Check DOES NOT deduct the cost of the request from the provided bucket's
+// capacity. The returned *Decision indicates whether the capacity exists to
+// satisfy the cost and represents the hypothetical state of the bucket IF the
+// cost WERE to be deducted. If no bucket exists it will NOT be created. No
+// state is persisted to the underlying datastore.
+func (l *Limiter) Check(ctx context.Context, bucket BucketWithCost) (*Decision, error) {
+	if bucket.cost < 0 {
```

Comment:
Having an array of objects which pair the bucket and the cost to be spent/ refunded really simplifies the batched rate limit decisions I'm adding in the next two PRs. 

For instance, I can construct an array of 3 `CertificatesPerDomain`, each with `cost = 1` along with an additional `CertificatesPerDomainPerAccount` with `cost = 3`, when an override is present. Then make a combined decision for the batch.

Reply:
I think `BucketWithCost` (or maybe `BucketAndCost`, if you want to make it clearer that this object holds both a bucket and a cost, rather than the bucket holding the cost?) is totally fine for the name of the type, my only nit is with the naming of the _variable_. Calling the local variable `bucket`, and then accessing the field `bucket.cost`, does make it seem like the bucket itself has a cost, rather than this object having both a bucket and a cost.

Reply:
Okay, I _think_ I've got this sorted out.